docs/vm-installation.md

```diff
@@ -0,0 +1,114 @@
+# Installing antrea-agent on VM
+
+`antrea-agent` can be run on Linux/Windows VMs. This document describes the steps needed to configure and run `antrea-agent`
+on VMs.
+
+## Prerequisites on Kubernetes cluster
+
+1. Enable `ExternalNode` feature on the `antrea-controller`
+2. Create a Namespace for `antrea-agent`. This document will use vm-ns as a NameSpace example for illustration.
+
+```bash
+kubectl create ns vm-ns
+```
+
+3. Create a ServiceAccount, ClusterRole and ClusterRoleBinding for `antrea-agent`. Update the [RBAC file](../build/yamls/externalnode/vm-agent-rbac.yml)
+   with the NameSpace created in the previous step and apply the RBAC file.
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/antrea-io/antrea/feature/externalnode/build/yamls/externalnode/vm-agent-rbac.yml
+```
+
+4. Create `antrea-agent.kubeconfig` file to access K8S API server
+
+```bash
+export CLUSTER_NAME=<cluster-name>
+export SERVICE_ACCOUNT=<service-account-name>
+APISERVER=$(kubectl config view -o jsonpath="{.clusters[?(@.name==\"$CLUSTER_NAME\")].cluster.server}")
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-cluster kubernetes --server=$APISERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-context antrea-agent@kubernetes --cluster=kubernetes --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.kubeconfig use-context antrea-agent@kubernetes
+```
+
+5. Create `antrea-agent.antrea.kubeconfig` file to access `antrea-controller` API server. VM should have network
+   connectivity to reach antrea-controller server
+
+```bash
+export ANTREA_API_SERVER=<Antrea API service endpoint>
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-cluster antrea --server=$ANTREA_API_SERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-context antrea-agent@antrea --cluster=antrea --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig use-context antrea-agent@antrea
+```
+
+## Prerequisites on Linux VM
+
+OVS needs to be installed. Please refer to the [getting-started guide](getting-started.md#open-vswitch)
+
+## Running antrea-agent on Linux VM
+
+1. Generate `antrea-agent` binaries
```

Comment:
I have few more changes to be added in installation doc. I will make the change and send it out for review.
I will create an another PR only for documentation, as it can go independently of other patches. Makes it easier to review/update the PR.

Reply:
A separate PR for documentation makes sense to me.